improver_tests/wxcode/wxcode/test_WeatherSymbols.py

```diff
@@ -823,7 +853,19 @@ def test_lightning(self):
         expected_wxcode[0, 1] = 29
         expected_wxcode[1, 1:] = 29
         expected_wxcode[2, 0] = 29
-        self.assertArrayEqual(result.data, expected_wxcode)
+        self.assertArrayAndMaskEqual(result.data, expected_wxcode)
+
+    def test_masked_precip(self):
+        """Test process returns right values when precipitation data are masked
+        (e.g. nowcast-only). """
+        plugin = WeatherSymbols()
+        data_precip = np.ma.masked_all_like(self.cubes[8].data)
+        cubes = self.cubes
+        cubes[8].data = data_precip
+        result = plugin.process(self.cubes)
+        expected_wxcode = np.ma.masked_all_like(self.expected_wxcode)
+        expected_wxcode[0, 1] = self.expected_wxcode[0, 1]
+        self.assertArrayAndMaskEqual(result.data, expected_wxcode)
```

Comment:
Why is there unmasked data at one point, when all the input points are masked.

Reply:
Because I only masked one of the input cubes. There is a pathway through the weather code tree that doesn't need this masked cube, so a weather code can be derived. Writing this explanation suggests that this isn't quite the right behaviour. I intended that if ANY pathway hits missing data, then the result would be missing, but I think I have if ALL pathways hit missing data, the result is missing. I'll look at this next week.

Reply:
Thinking about this a bit more, I'm happy that the code is behaving as I want it to. Only one route through the tree can ever be "True" and if that route contains a missing data value, then the result will be a missing data value. I've updated the doc-string of this test to describe how the lightning symbol can still be achieved (it doesn't depend on precipitation).